src/System.Numerics.Tensors/System/Numerics/Tensors/Tensor.cs

```diff
@@ -124,224 +124,137 @@ public static Tensor<T> CreateFromDiagonal<T>(Tensor<T> diagonal, int offset)
 
             return result;
         }
+    }
 
-        internal static void ValidateBinaryArgs<T>(Tensor<T> left, Tensor<T> right)
+    /// <summary>
+    /// Represents a multi-dimensional collection of objects of type T that can be accessed by indices.
+    /// </summary>
+    /// <typeparam name="T">type contained within the Tensor.  Typically a value type such as int, double, float, etc.</typeparam>
+    [DebuggerDisplay("{GetArrayString(false)}")]
+    // When we cross-compile for frameworks that expose ICloneable this must implement ICloneable as well.
+    public abstract class Tensor<T> : IList, IList<T>, IReadOnlyList<T>, IStructuralComparable, IStructuralEquatable
+    {
+        internal static T Zero
```

Comment:
I wonder if we can simplify this by returning default(T), if that returns the same in all cases I'd opt for that.

Reply:
I did that originally, but it doesn't cover the case where `T` is an unsupported type.

Reply:
I’d be ok with permitting that.  I think that’d let us do sparse reference types (string)